Enable OOB tests

welansari opened the pull request:
- Enable OOB tests
- Increase the timeout of OOB tests as replication of operations between the repds can be slow when the CI vm is saturated (an operation is only visible in the raft log once it got replicated to the majority of the repds)
- Wait for the ingestion consumer group to be in a stable state before starting the tests. Zenko is reconciled just before launching the tests which causes the ingestion pods to restart and the consumer group to rebalance. The consumer group stays in the rebalance state for 30-ish seconds which is enough time for the tests to start. Not waiting for the state to be stable can cause the first messages pushed to the topic to be skipped (consumers start consuming late because of the rebalance) as the consumers are configured to start consuming from the latest offset if no offset was found (which is the case at startup).
- When we wait for the consumer group state to be stable we also check if we have the proper number of consumers connected to the consumer group. When the ingestion pods restart from the reconcile, the previous consumers are kept in the consumer group until their session times out. Not waiting for this can cause messages to be skipped as well, as the new consumer(s) only have some partitions assigned to them and not all, the rest are assigned to the "dead" consumer until future rebalance. At startup no offset is generally stored yet for the partitions assigned to the "dead" consumer, so when the rebalance occurs and the new consumer get assigned to them the new offset becomes the latest and previous messages are skipped.

Issue: ZENKO-4286

Commit:
enable OOB tests

Issue: ZENKO-4286

Commit:
set backbeat logging level to debug

Issue: ZENKO-4286

bert-e commented:
# Hello kerkesni,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4286 contains:


* `2.11.0`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.10.12`


Please check the `Fix Version/s` of ZENKO-4286, or the target
branch of this pull request.

Commit:
increase oob test timeout

A raft log entry is considered committed only after
it has been replicated to the majority of the repds.

Increasing OOB test timeout to allow this replication
to happen.

Issue: ZENKO-4286

Commit:
wait for the ingestion processor to start consuming from Kafka

Zenko gets reconciled just before launching the tests which makes
the first two ingestion tests falky as the consumer group is not
stable (rebalance state) when the tests are started. This
causes the consumers to connect late to the topic (after the producer
pushed the first message), which causes the first messages to be skipped
(consumers are configured to start consuming from the latest offset when
no offset is stored in Kafka)

Issue: ZENKO-4286

Commit:
do not check return code in oob test

"data" is the actual body of the response which
does not contain the error code.

Issue: ZENKO-4286

Commit:
wait to have the expected number of consumers in the consumer group

When a pod restarts the previous consumers are kept in the consumer group
until the session times out, this can lead to the new consumer skipping
some messages on the non assigned partitions as there is no offset stored
for those initially (consumers are configured to start consuming from the
latest offset).

Issue: ZENKO-4286

Commit:
make namespace configurable in wait_for_consumer_group

Issue: ZENKO-4286

Commit:
dynamically get the ingestion consumer group expected member count

Issue: ZENKO-4286

Commit:
only wait for ingestion consumer group when ingestion tests are enabled

Issue: ZENKO-4286

francoisferrand commented on .github/scripts/end2end/configure-e2e.sh:
```diff
@@ -109,3 +109,10 @@ sleep 10
+
+
+if [ $ENABLE_RING_TESTS = true ]; then
+  # wait for ingestion processor to start consuming from Kafka
```
this should (eventually) get baked into backbeat?
--> so we should probably add a ticket (and we can put it in the EPIC "cleanup kafka management" https://scality.atlassian.net/browse/ARTESCA-9180)

welansari replied:
https://scality.atlassian.net/browse/BB-655

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

welansari commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.11`



The following branches will **NOT** be impacted:

* `development/2.10`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.11`



The following branches have **NOT** changed:

* `development/2.10`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue ZENKO-4286.

Goodbye kerkesni.
